accounts/admin.py

```diff
@@ -248,9 +255,45 @@ class OldUsernameAdmin(admin.ModelAdmin):
     list_display = ('user', 'username')
 
 
+class UserGDPRDeletionRequestAdmin(admin.ModelAdmin):
+    search_fields = ('=username', '=email')
+    raw_id_fields = ('user', )
+    list_display = ('email', 'username', 'user_link', 'deleted_user_link', 'status')
+    list_filter = ('status', )
+    fieldsets = (
+        (None, {'fields': ('email', 'status', 'user', 'username')}),
+    )
+
+    def get_queryset(self, request):
+        # overrride 'get_queryset' to optimize query by using select_related on 'user' and 'deleted_user'
+        qs = super(UserGDPRDeletionRequestAdmin, self).get_queryset(request)
+        qs = qs.select_related('user', 'deleted_user')
+        return qs
+
+    def user_link(self, obj):
+        return '<a href="{0}" target="_blank">{1}</a>'.format(reverse('admin:auth_user_change', args=[obj.user_id]),
+                                                              obj.user.username)
+    user_link.allow_tags = True
+    user_link.admin_order_field = 'user'
+    user_link.short_description = 'User object'
+
+    def deleted_user_link(self, obj):
+        if obj.deleted_user is None:
+            return '-'
+        return '<a href="{0}" target="_blank">{1}</a>'.format(reverse('admin:accounts_deleteduser_change', args=[obj.deleted_user_id]),
+                                                              obj.deleted_user.username)
+    deleted_user_link.allow_tags = True
+    deleted_user_link.admin_order_field = 'deleted_user'
+    deleted_user_link.short_description = 'Deleted user object'
+
+
+class DeletedUserAdmin(admin.ModelAdmin):
+    readonly_fields = ['user']
```

Comment:
I added this, otherwise the page fails to load (as it's trying to get all of the users to put in a dropdown for the user relation).
However, I think it might be a good idea to make _all_ of these fields readonly... what do you think? There should be no need to ever edit this object.

Reply:
Addressed in 0f346f6f1b5b4542a9282530962111bc54b3c1e8